ambry-account/src/main/java/com/github/ambry/account/MySqlAccountService.java

```diff
@@ -54,289 +45,97 @@
-    super(config, Objects.requireNonNull(accountServiceMetrics, "accountServiceMetrics cannot be null"), notifier);
+  private final MySqlAccountStore mySqlAccountStore;
+  private final CachedAccountService cachedAccountService;
+  private CachedAccountService cachedAccountServiceNew;
+
+  public MySqlAccountService(AccountServiceMetricsWrapper accountServiceMetrics, MySqlAccountServiceConfig config,
+      MySqlAccountStoreFactory mySqlAccountStoreFactory, Notifier<String> notifier) throws IOException {
+    super(config, Objects.requireNonNull(accountServiceMetrics.getAccountServiceMetrics(),
+        "accountServiceMetrics cannot be null"), notifier);
     this.config = config;
-    this.mySqlAccountStoreFactory = mySqlAccountStoreFactory;
-    try {
-      this.mySqlAccountStore = mySqlAccountStoreFactory.getMySqlAccountStore();
-    } catch (SQLException e) {
-      logger.error("MySQL account store creation failed", e);
-      // If it is a non-transient error like credential issue, creation should fail.
-      // Otherwise, continue account service creation and initialize cache with metadata from local file copy
-      // to serve read requests. Connection to MySql DB will be retried during periodic sync. Until then, write
-      // requests will be blocked.
-      if (MySqlDataAccessor.isCredentialError(e)) {
-        // Fatal error, fail fast
-        throw e;
-      }
-    }
-    accountServiceMetrics.trackTimeSinceLastSync(this::getTimeInSecondsSinceLastSync);
-    accountServiceMetrics.trackContainerCount(this::getContainerCount);
     this.scheduler =
         config.updaterPollingIntervalSeconds > 0 ? Utils.newScheduler(1, MYSQL_ACCOUNT_UPDATER_PREFIX, false) : null;
-    // create backup file manager for persisting and retrieving Account metadata on local disk
-    backupFileManager = new BackupFileManager(accountServiceMetrics, config.backupDir, config.maxBackupFileCount);
-    // A local LRA cache of containers not found in mysql db. This is used to avoid repeated queries to db during getContainerByName() calls.
-    recentNotFoundContainersCache = Collections.newSetFromMap(
-        Collections.synchronizedMap(new LinkedHashMap<String, Boolean>(cacheInitialCapacity, cacheLoadFactor, true) {
-          protected boolean removeEldestEntry(Map.Entry<String, Boolean> eldest) {
-            return size() > cacheMaxLimit;
-          }
-        }));
-    this.writeCacheAfterUpdate = config.writeCacheAfterUpdate;
-    // Initialize cache from backup file on disk
-    initCacheFromBackupFile();
-    // Fetches added or modified accounts and containers from mysql db and schedules to execute it periodically
-    initialFetchAndSchedule();
+    this.mySqlAccountStore = callSupplierWithException(mySqlAccountStoreFactory::getMySqlAccountStore).get();
+    cachedAccountService =
+        new CachedAccountService(callSupplierWithException(mySqlAccountStoreFactory::getMySqlAccountStore),
```

Comment:
I have to use the supplier since it's used for lazy connection when fetchAndUpdateCache(). But I just realize that for the first initialization, I need to only throw exception when it's failed due to CredentialError. So I updated the supplier a little bit to always through exception, and for initialization, I'll initialize in mysqlAccountService constructor and pass it in.